target/linux/pistachio/base-files/lib/pistachio.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+PISTACHIO_BOARD_NAME=
+PISTACHIO_MODEL=
+
+pistachio_board_detect() {
+	local name=$(grep "^machine" /proc/cpuinfo | sed "s/machine.*: IMG \(.*\) (\(.*\))/\1/g" | awk '{print tolower($1)}')
+	local model=$(grep "^machine" /proc/cpuinfo | sed "s/machine.*: IMG \(.*\) (\(.*\))/\2/g")
```

Comment:
Managed to delete my comment, sorry. Anyhow after reviewing what every other target does I changed it to work the same way. Please take a look at the latest version, do you feel it would still be best to make the changes you have suggested?